src/googletest.h

```diff
@@ -574,7 +574,8 @@ class Thread {
 
 static inline void SleepForMilliseconds(int t) {
 #ifndef OS_WINDOWS
-  usleep(t * 1000);
+  const struct timespec req = {0, t * 1000 * 1000};
+  nanosleep(&req, NULL);
```

Comment:
Don't you have to check for `_POSIX_C_SOURCE >= 199309L`?

By the way, this change should be particularly submitted to the upstream (if not done yet).

Reply:
What upstream?

Reply:
If you mean googletest, it's already using nanosleep: https://github.com/google/googletest/blob/master/googletest/include/gtest/internal/gtest-port.h#L1138

(strange as C++11 has sleep_for for exactly this purpose).

Reply:
wait a minute...even this is C++.